public/app/plugins/panel/geomap/components/DataHoverView.tsx

```diff
@@ -17,45 +17,71 @@ export class DataHoverView extends PureComponent<Props> {
 
   render() {
     const { data, feature, rowIndex, columnIndex } = this.props;
-
-    if (feature) {
-      return (
-        <table className={this.style.infoWrap}>
-          <tbody>
-            {Object.entries(feature.getProperties()).map(
-              (e, i) =>
-                e[0] === 'geometry' || ( //don't include geojson feature geometry
-                  <tr key={`${e}-${i}`}>
-                    <th>{`${e[0]}: `}</th>
-                    <td>{`${e[1]}`}</td>
-                  </tr>
-                )
-            )}
-          </tbody>
-        </table>
-      );
-    }
-
-    if (!data || rowIndex == null) {
-      return null;
-    }
-
+    const excludeFields = ['geometry']; //TODO: add custom fields
+    const details = getDetails({ data, feature, rowIndex, columnIndex, excludeFields });
     return (
       <table className={this.style.infoWrap}>
         <tbody>
-          {data.fields.map((f, i) => (
-            <tr key={`${i}/${rowIndex}`} className={i === columnIndex ? this.style.highlight : ''}>
-              <th>{getFieldDisplayName(f, data)}:</th>
-              <td>{fmt(f, rowIndex)}</td>
-            </tr>
-          ))}
+          {details &&
+            details.map((d, i) => {
+              return (
+                <tr key={`${i}-${d}`} className={i === columnIndex ? this.style.highlight : ''}>
+                  <th>{`${d.header}`}</th>
+                  <td>{`${d.data}`}</td>
+                </tr>
+              );
+            })}
         </tbody>
       </table>
     );
   }
 }
 
-function fmt(field: Field, row: number): string {
+const getDetails = (
+  props: Partial<Props> & { excludeFields: string[] }
+): Array<{ header: string; data: string }> | null => {
+  const { data, feature, rowIndex, excludeFields } = props;
+  if (feature) {
+    const features = feature.get('features');
```

Comment:
I am a bit concerned that the clustering logic lives in here, but it is likely OK.

Perhaps (no now!) we could attach a function like this to the hover event and format on demand :man_shrugging:

Reply:
Definitely - wasn't planning on this being a long-term fix but more of a band-aid to help with seeing how many features are represented at that point. 

Eventually want to use a specific style on hover - so hoping to investigate more about map interactions sometime soon.